core/include/io.h

```diff
@@ -106,18 +106,56 @@ static inline void put_be16(void *p, uint16_t val)
  */
 static inline void io_setbits32(uintptr_t addr, uint32_t set_mask)
 {
-	write32(read32(addr) | set_mask, addr);
+	io_write32(addr, io_read32(addr) | set_mask);
 }
 
 static inline void io_clrbits32(uintptr_t addr, uint32_t clear_mask)
 {
-	write32(read32(addr) & ~clear_mask, addr);
+	io_write32(addr, io_read32(addr) & ~clear_mask);
 }
 
 static inline void io_clrsetbits32(uintptr_t addr, uint32_t clear_mask,
 				   uint32_t set_mask)
 {
-	write32((read32(addr) & ~clear_mask) | set_mask, addr);
+	io_write32(addr, (io_read32(addr) & ~clear_mask) | set_mask);
+}
+
+/*
+ * Functions write8(), write16(), write32(), read8(), read16() and read32()
+ * will be deprecated in OP-TEE release 3.5.0.
```

Comment:
How about deprecating them immediately and remove them in another commit in this PR?
Other trees can then choose if they want to pick up the removing of `write{8,16,32}()` or how long to delay it.

Reply:
Ok, I will update this P-R with appended commit for core/drivers/patforms update and deprecation (function removal).